Ansible declarations

DaMandal0rian opened the pull request:
- Add docker compose deployments for networks
- user ssh deployment
- secure server setup
- base installation setup
- monitoring packages
- CI workflows for ansible deployment
- add transcrypt to infra repo for encrypting host files

closes #165

DaMandal0rian committed:
Ansible setup

DaMandal0rian committed:
remove monitoring stack

- remove the monitoring stack since using docker compose instead

DaMandal0rian committed:
minor changes

- minor fix nginx
- Firewall and host adjustments
- add ssh keys for users

DaMandal0rian committed:
add network deployment with ansible

- add ansible deployment docker compose
- add github ci action

DaMandal0rian committed:
add docker compose files and var files

remove plaintext host files

DaMandal0rian committed:
add transcrypt with encrypted host file

DaMandal0rian committed:
add and fix some docker manifests

- bump webfactory actions version

vedhavyas commented on ansible/ephemeral-devnets/files/docker-compose-bootstrap-devnet.yml:
```diff
@@ -0,0 +1,74 @@
+      "--chain", "${NETWORK_NAME}",
+      "--base-path", "/var/subspace",
+      "--execution", "wasm",
+#      "--enable-subspace-block-relay",
```
can we keep this under flag or remove it if not required

DaMandal0rian replied:
sure

vedhavyas commented on ansible/ephemeral-devnets/files/docker-compose-node-devnet.yml:
```diff
@@ -0,0 +1,61 @@
+
+services:
+  # caddy reverse proxy with automatic tls management using let encrypt
+  caddy:
```
I thought we are not using caddy anymore and instead used traefik.

DaMandal0rian replied:
caddy is used in the docker compose, we don't use traefik here, it's used in logging stack. Later will look into using traefik or nginx everywhere but it's not high priority right now.

vedhavyas commented on ansible/ephemeral-devnets/playbooks/deploy-compose.yml:
```diff
@@ -0,0 +1,12 @@
+  tasks:
+    - name: Copy Docker Compose file to the target server
+      copy:
+        src: "../files/{{ docker_compose_file }}"
```
Why not use the ansible docker compose module instead ?
https://docs.ansible.com/ansible/latest/collections/community/docker/docker_compose_module.html

DaMandal0rian replied:
Because the ansible module does not support compose v2, and still on-going work to do it. Docker compose v2 is based on Go while version 1 was python.

vedhavyas commented on ansible/network/devnet/hosts.yml:
```diff
@@ -0,0 +1,23 @@
+U2FsdGVkX18a5DLamhYs7yLNDfiMoamNZRhAWfKT3lDYSM+3CGh5Gp40G36c7g3w
```
IIUC, these should be the IPs of the hosts, no ?

DaMandal0rian replied:
IPs and hostnames of the hosts, they are but encrypted.

vedhavyas commented on ansible/server-setup/ansible.cfg:
```diff
@@ -0,0 +1,4 @@
+[defaults]
+INVENTORY = hosts
+
+ansible_shell_type=zsh
```
zsh is not default, bash is always preferred

DaMandal0rian replied:
Sure, will use bash if it's preferred.

vedhavyas commented on ansible/server-setup/roles/netdata/vars/main.yml:
```diff
@@ -0,0 +1,29 @@
+# Copy those strings into the variables below. `claim_url` should be
+# `https://app.netdata.cloud`. Read more:
+# https://learn.netdata.cloud/docs/agent/claim
+claim_token: XXXXX
```
Did you forget to update these? Should they come from the env instead ?

DaMandal0rian replied:
No, this is a variable file, and the tokens aren't fixed. Need to update the values before running the playbook, or passing the `-e` or `--extra-vars` flags when running the playbook to override the default value.

vedhavyas commented on ansible/server-setup/roles/nginx/files/nginx.conf:
```diff
@@ -0,0 +1,127 @@
+user www-data;
+worker_processes 16;
+worker_rlimit_nofile 32000;
+pid /run/nginx.pid;
```
Why do we need to nginx here ?

DaMandal0rian replied:
Because some servers use nginx such as block explorer infra and blockscout

vedhavyas commented on ansible/server-setup/roles/security/tasks/main.yml:
```diff
@@ -0,0 +1,67 @@
+    state: restarted
+  changed_when: false
+
+- name: Enable firewall
```
Shouldn't we enable firewall first and then set rules after ?

DaMandal0rian replied:
No, the firewall can be enabled as a last step.

vedhavyas commented on ansible/server-setup/roles/setup/tasks/install_zsh.yml:
```diff
@@ -0,0 +1,4 @@
+---
+- name: Install ohmyzsh shell
```
Lets not install ohmyzsh on servers pls

DaMandal0rian replied:
sure, removed it.

vedhavyas reviewed (commented):
Left some questions. I think we are not ansible to full potential. We are still using docker-compose setup on the server instead of using ansible module. I think we should use that

DaMandal0rian committed:
changes to docker and setup packages

- remove zsh and vector as won't be used
- docker compose manifests formatting
- remove unused flag --enable-subspace-block-relay from docker compose
